docs/README.md

```diff
@@ -38,181 +38,92 @@ Code | Text | Description
-  'checkout-timestamp': '2018-07-06T10:01:31.904Z'
-};
-
-const body = {
-  stamp: 'should-be-unique-for-merchant',
-  reference: '3759170',
-  amount: 1525,
-  currency: 'EUR',
-  language:'FI',
-  items: [
-    {
-      unitPrice: 1525,
-      units: 1,
-      vatPercentage: 24,
-      productCode: '#1234',
-      deliveryDate: '2018-09-01'
-    }
-  ],
-  customer: {
-    email: 'test.customer@example.com'
-  },
-  redirectUrls: {
-    success: 'https://ecom.example.com/cart/success',
-    cancel: 'https://ecom.example.com/cart/cancel'
-  }
-};
-
-const hmacPayload =
-  Object.keys(headers)
-    .sort()
-    .map((key) => [ key, headers[key] ].join(':'))
-    .concat(JSON.stringify(body))
-    .join("\n");
-
-// Expected HMAC: e6ed7ec0889db888f3067feb57e0a831b88da547902cd4f40ecb646d2bb763ac
-const hmac = crypto
-  .createHmac('sha256', SECRET)
-  .update(hmacPayload)
-  .digest('hex');
+```
+checkout-account:1234\n
+checkout-algorithm:sha256\n
+checkout-method:POST\n
+checkout-nonce:1234\n
+checkout-timestamp:2018-07-05T11:19:25.950Z\n
+REQUEST BODY
 ```
 
+See also code examples [HMAC calculation (node.js)](/examples#hmac-calculation-node-js) and [HMAC calculation (PHP)](/examples#hmac-calculation-php).
 
-### HMAC example (PHP)
-
-```php
-<?php
-
-$ACCOUNT = '375917';
-$SECRET = 'SAIPPUAKAUPPIAS';
-
-$headers = array(
-    'checkout-account' => $ACCOUNT,
-    'checkout-algorithm' => 'sha256',
-    'checkout-method' => 'POST',
-    'checkout-nonce' => '564635208570151',
-    'checkout-timestamp' => '2018-07-06T10:01:31.904Z'
-);
-
-// Headers must be sorted alphabetically by their key
-ksort($headers);
-
-$body = array(
-    'stamp' =>  'should-be-unique-for-merchant',
-    'reference' => '3759170',
-    'amount' => 1525,
-    'currency' => 'EUR',
-    'language' => 'FI',
-    'items' => array(
-        array(
-            'unitPrice' => 1525,
-            'units' => 1,
-            'vatPercentage' => 24,
-            'productCode' => '#1234',
-            'deliveryDate' => '2018-09-01'
-        )
-    ),
-    'customer' => array(
-        'email' => 'test.customer@example.com'
-    ),
-    'redirectUrls' => array(
-        'success' => 'https://ecom.example.com/cart/success',
-        'cancel' => 'https://ecom.example.com/cart/cancel'
-    )
-);
-
-
-$headers =
-    array_map(
-        function ($val, $key) {
-            return $key . ':' . $val;
-        },
-        $headers,
-        array_keys($headers)
-    );
-
-array_push($headers, json_encode($body, JSON_UNESCAPED_SLASHES));
-
-// string(64) "e6ed7ec0889db888f3067feb57e0a831b88da547902cd4f40ecb646d2bb763ac"
-$hmac = hash_hmac('sha256', join("\n", $headers), $SECRET);
-```
+#### Redirect and callback URL signing
 
+Return and callback URL parameters are signed as well, and the merchant *must* check signature validity. The signature is calculated the same way as for requests, but the values come in as query string parameters instead of headers. Empty string is used for the body.
```

Comment:
Additional details that I would add.

Under either separate header or just add it after the paragraph.

The returns urls and callbacks need to support the fact that PSP might call them in a later time and multiple times. So the callback needs to contain the actual unique identifying information of the payment on top of that